src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -33,11 +42,66 @@ def cli_asset():
     default=False,
 )
 @_handle_error
-def asset_snapshot(**args):
+def asset_snapshot(root_dir, manifest_out, recursive, **args):
     """
     Creates manifest of files specified root directory.
     """
-    click.echo("snapshot taken")
+    root_dir_basename = os.path.basename(root_dir) + "_"
+
+    if not os.path.isdir(root_dir):
+        misconfigured_directories_msg = f"Specified root directory {root_dir} does not exist. "
+        raise NonValidInputError(misconfigured_directories_msg)
+
+    if manifest_out and not os.path.isdir(manifest_out):
+        misconfigured_directories_msg = (
+            f"Specified destination directory {manifest_out} does not exist. "
+        )
+        raise NonValidInputError(misconfigured_directories_msg)
+    elif manifest_out is None:
+        manifest_out = root_dir
+
+    inputs = []
+    for root, dirs, files in os.walk(root_dir):
+        if os.path.basename(root).endswith("_manifests"):
+            continue
+        for file in files:
+            file_full_path = str(os.path.join(root, file))
+            inputs.append(file_full_path)
```

Comment:
nit: this can be simplified to:
```suggestion
        inputs.extend(files)
```